src/Symfony/Component/Workflow/Registry.php

```diff
@@ -23,19 +24,26 @@ class Registry
 
     /**
      * @param Workflow $workflow
-     * @param string   $className
+     * @param SupportStrategyInterface $supportStrategy
      */
-    public function add(Workflow $workflow, $className)
+    public function add(Workflow $workflow, SupportStrategyInterface $supportStrategy)
```

Comment:
True, one could do that to avoid BC, but it's introducing kind of magic below, isn't it?

I ike the deprecation idea though 👍